### PR TITLE
[INLONG-11605][Audit] Added audit data legitimacy verification

### LINKED ISSUE #11605: [Improve][Audit] Added audit data legitimacy verification
### Description

Abnormal data timestamps, illegal inlong_group_id and inlong_stream_id may affect the audit, and these abnormal information need to be filtered

### InLong Component

InLong Audit

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by doleyzi**
- Fixes #11605

### Motivation

Added audit data legitimacy verification

### Modifications

Verify the legitimacy of the dimensions and indicators of audit data, discard dirty data, and report to monitoring

### CODE REVIEW

File: inlong-audit/audit-common/src/main/java/org/apache/inlong/audit/utils/DataUtils.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        return timeDiff <= deviation;
+    }
+
+    /**
+     * Checks if the audit item is valid.
+     *
+     * @param auditItem The audit item to check.
+     * @return true if the audit item is blank or its length is less than the maximum length, false otherwise.
+     */
+    public static boolean isAuditItemValid(String auditItem) {
```
Comment (aloyszhang) [inline inlong-audit/audit-common/src/main/java/org/apache/inlong/audit/utils/DataUtils.java:45]: The method name and the logic do not match.

File: inlong-audit/audit-store/src/main/java/org/apache/inlong/audit/store/utils/PulsarUtils.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.slf4j.LoggerFactory;
+
+public class PulsarUtils {
+
+    private static final Logger LOG = LoggerFactory.getLogger(PulsarUtils.class);
+    public static void acknowledge(Consumer<byte[]> consumer, MessageId messageId) {
+        try {
+            consumer.acknowledge(messageId);
+        } catch (Exception exception) {
+            LOG.error("Acknowledge has exception ", exception);
```
Comment (aloyszhang) [inline inlong-audit/audit-store/src/main/java/org/apache/inlong/audit/store/utils/PulsarUtils.java:32]: Better to print the topic, subscription and messageId

File: inlong-audit/audit-common/src/main/test/java/org/apache/inlong/audit/utils/DataUtilsTest.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+import org.junit.Test;
+
+import java.util.Random;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+
+public class DataUtilsTest {
```
Comment (aloyszhang) [inline inlong-audit/audit-common/src/main/test/java/org/apache/inlong/audit/utils/DataUtilsTest.java:28]: Why the test class is under the "main" directory?

File: inlong-audit/audit-store/src/main/java/org/apache/inlong/audit/store/utils/PulsarUtils.java (hunk 2 above)
Comment (doleyzi) [reply inlong-audit/audit-store/src/main/java/org/apache/inlong/audit/store/utils/PulsarUtils.java]: Fixed

**Review by justinwwhuang — APPROVED** [review]

**Review by aloyszhang — APPROVED** [review]

**Pull request merged**